apiserver/getdistrosigimageconfig.go

```diff
@@ -22,46 +23,36 @@ func (api *APIServer) GetDistroSigImageConfig(w http.ResponseWriter, r *http.Req
 	ctx, cancel := context.WithTimeout(ctx, 30*time.Second)
 	defer cancel()
 
-	processResult := make(chan Result)
-	go func() {
-		var config datamodel.GetLatestSigImageConfigRequest
+	var config datamodel.GetLatestSigImageConfigRequest
 
-		err := json.NewDecoder(r.Body).Decode(&config)
-		if err != nil {
-			handleError(err)
-			return
-		}
-
-		agentBaker, err := agent.NewAgentBaker()
-		if err != nil {
-			processResult <- handleError(err)
-			return
-		}
-
-		allDistros, err := agentBaker.GetDistroSigImageConfig(config.SIGConfig, config.Region)
-		if err != nil {
-			processResult <- handleError(err)
-			return
-		}
+	err := json.NewDecoder(r.Body).Decode(&config)
+	if err != nil {
+		log.Println(err.Error())
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
 
-		result, err := json.Marshal(allDistros)
-		if err != nil {
-			processResult <- handleError(err)
-			return
-		}
-		processResult <- Result{string(result), nil}
-	}()
+	agentBaker, err := agent.NewAgentBaker()
+	if err != nil {
+		log.Println(err.Error())
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
 
-	select {
-	case <-ctx.Done():
-		http.Error(w, "process timeout", http.StatusInternalServerError)
-	case result := <-processResult:
-		if result.Error != nil {
-			http.Error(w, result.Error.Error(), http.StatusBadRequest)
-		} else {
-			w.WriteHeader(http.StatusOK)
-			fmt.Fprint(w, result.Message)
-		}
+	allDistros, err := agentBaker.GetDistroSigImageConfig(config.SIGConfig, config.Region)
+	if err != nil {
+		log.Println(err.Error())
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
 
+	result, err := json.Marshal(allDistros)
+	if err != nil {
+		log.Println(err.Error())
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
 	}
+
+	w.WriteHeader(http.StatusOK)
+	fmt.Fprint(w, string(result))
```

Comment:
not important for this pr at all (in fact please don't make this change but you know you can deseriize directly to w with json.Encoder.